src/main.cpp

```diff
@@ -99,14 +106,12 @@ int main(int argc, char **argv)
     delete lammps;
   } catch (fmt::format_error &fe) {
     fprintf(stderr, "fmt::format_error: %s\n", fe.what());
-    KokkosLMP::finalize();
-    Python::finalize();
+    finalize();
     MPI_Abort(MPI_COMM_WORLD, 1);
     exit(1);
   }
 #endif
```

Comment:
Because there is now a try catch block even for the case without `LAMMPS_EXCEPTIONS` couldn't we write the whole block as:
```cpp
  try {
    auto lammps = new LAMMPS(argc, argv, lammps_comm);
    lammps->input->file();
    delete lammps;
#ifdef LAMMPS_EXCEPTIONS
  } catch (LAMMPSAbortException &ae) {
    finalize();
    MPI_Abort(ae.universe, 1);
  } catch (LAMMPSException &) {
    finalize();
    MPI_Barrier(lammps_comm);
    MPI_Finalize();
    exit(1);
#endif
  } catch (fmt::format_error &fe) {
    fprintf(stderr, "fmt::format_error: %s\n", fe.what());
    finalize();
    MPI_Abort(MPI_COMM_WORLD, 1);
    exit(1);
  } catch (std::exception &e) {
    fprintf(stderr, "Exception: %s\n", e.what());
    finalize();
    MPI_Abort(MPI_COMM_WORLD, 1);
    exit(1);
  }
```
feel a bit more natural, saves some code duplication and catches exception from any other third party library.

But feel free to ignore.

Reply:
I have a more ambitious plan, please see issue #3678. But this is something for after the next stable release.